crates/oxc_semantic/src/builder.rs

```diff
@@ -394,6 +413,1902 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
+    }
+
+    fn visit_ts_signature(&mut self, signature: &TSSignature<'a>) {
+        match &signature {
+            TSSignature::TSPropertySignature(sig) => self.visit_ts_property_signature(sig),
+            TSSignature::TSCallSignatureDeclaration(sig) => {
+                self.visit_ts_call_signature_declaration(sig);
+            }
+            TSSignature::TSIndexSignature(sig) => self.visit_ts_index_signature(sig),
+            TSSignature::TSMethodSignature(sig) => self.visit_ts_method_signature(sig),
+            TSSignature::TSConstructSignatureDeclaration(sig) => {
+                self.visit_ts_construct_signature_declaration(sig);
+            }
+        }
+    }
+
+    fn visit_ts_construct_signature_declaration(
+        &mut self,
+        signature: &TSConstructSignatureDeclaration<'a>,
+    ) {
+        self.visit_formal_parameters(&signature.params);
+        if let Some(parameters) = &signature.type_parameters {
+            self.visit_ts_type_parameter_declaration(parameters);
+        }
+        if let Some(annotation) = &signature.return_type {
+            self.visit_ts_type_annotation(annotation);
+        }
+    }
+
+    fn visit_ts_method_signature(&mut self, signature: &TSMethodSignature<'a>) {
+        let kind = AstKind::TSMethodSignature(self.alloc(signature));
+        self.enter_node(kind);
+        self.visit_formal_parameters(&signature.params);
+        if let Some(parameters) = &signature.type_parameters {
+            self.visit_ts_type_parameter_declaration(parameters);
+        }
+        if let Some(annotation) = &signature.return_type {
+            self.visit_ts_type_annotation(annotation);
+        }
+        self.leave_node(kind);
+    }
+
+    fn visit_ts_index_signature_name(&mut self, name: &TSIndexSignatureName<'a>) {
+        self.visit_ts_type_annotation(&name.type_annotation);
+    }
+
+    fn visit_ts_index_signature(&mut self, signature: &TSIndexSignature<'a>) {
+        for name in &signature.parameters {
+            self.visit_ts_index_signature_name(name);
+        }
+
+        self.visit_ts_type_annotation(&signature.type_annotation);
+    }
+
+    fn visit_ts_property_signature(&mut self, signature: &TSPropertySignature<'a>) {
+        let kind = AstKind::TSPropertySignature(self.alloc(signature));
+        self.enter_node(kind);
+        self.visit_property_key(&signature.key);
+        if let Some(annotation) = &signature.type_annotation {
+            self.visit_ts_type_annotation(annotation);
+        }
+        self.leave_node(kind);
+    }
+
+    fn visit_ts_call_signature_declaration(&mut self, signature: &TSCallSignatureDeclaration<'a>) {
+        self.visit_formal_parameters(&signature.params);
+        if let Some(parameters) = &signature.type_parameters {
+            self.visit_ts_type_parameter_declaration(parameters);
+        }
+
+        if let Some(annotation) = &signature.return_type {
+            self.visit_ts_type_annotation(annotation);
+        }
+    }
```

Comment:
Can you remove all the visitors that don't have a cfg override right now? Or are you changing the visit order in all these visits?

Reply:
Done.